nested-types.ipynb

```diff
@@ -0,0 +1,915 @@
+       "    - dictionary: NULL\n",
+       "    - children[0]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'int8'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[2]:\n",
+       "      - validity <bool[1 b] 11110000>\n",
+       "      - data <int8[5 b] 1 2 0 4 0>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "na.c_array_view(na.c_array(column_4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Map\n",
+    "\n",
+    "Map type represents nested data where each value is a variable number of key-item pairs. Its physical representation is the same as a list of `{key, item}` structs.\n",
+    "\n",
+    "The difference between a struct and a map type is that a struct holds the key in the schema therefore needs to be a string and the items are all equal type and stored in values buffer per each field. There can be multiple keys and therefore multiple values buffers. The map comparing to the struct has one child array holding all different keys that need to be all same type but not necessarily string and second child array holding all the values which also need to be the same type.\n",
+    "\n",
+    "Also, the map stores the struct in a list and needs an offset as the list is variable shape.\n",
+    "\n",
+    "![image info](./diagrams/map-diagram.svg)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<pyarrow.lib.MapArray object at 0x11126f8e0>\n",
+       "[\n",
+       "  keys:\n",
+       "  [\n",
+       "    \"Dark Knight\"\n",
+       "  ]\n",
+       "  values:\n",
+       "  [\n",
+       "    10\n",
+       "  ],\n",
+       "  keys:\n",
+       "  [\n",
+       "    \"Dark Knight\",\n",
+       "    \"Meet the Parents\",\n",
+       "    \"Superman\"\n",
+       "  ]\n",
+       "  values:\n",
+       "  [\n",
+       "    8,\n",
+       "    4,\n",
+       "    5\n",
+       "  ],\n",
+       "  null,\n",
+       "  keys:\n",
+       "  [\n",
+       "    \"Meet the Parents\",\n",
+       "    \"Superman\"\n",
+       "  ]\n",
+       "  values:\n",
+       "  [\n",
+       "    10,\n",
+       "    null\n",
+       "  ]\n",
+       "]"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_6_data = [{'Dark Knight': 10},\n",
+    "                 {'Dark Knight': 8, 'Meet the Parents': 4, 'Superman': 5},\n",
+    "                 None,\n",
+    "                 {'Meet the Parents': 10, 'Superman': None}]\n",
+    "column_6 = pa.array(column_6_data, type=pa.map_(pa.string(), pa.int32(), keys_sorted=True))\n",
```

Comment:
```suggestion
    "column_6 = pa.array(column_6_data, type=pa.map_(pa.string(), pa.int32()))\n",
```

Given that this is an optional keyword, I would leave that out to keep it a bit simpler